lib/src/runner/reporter/xunit.dart

```diff
@@ -0,0 +1,363 @@
+            testFailure.stack.split('\n').forEach((line) {
+              individualTest += '\n' + ' ' * (counter + 6) + line;
+            });
+            individualTest += '\n' + ' ' * (counter + 4) + '</failure>';
+          });
+        } else {
+          test.error['list'].forEach((XunitFailureResult testError) {
+            individualTest += '\n' +
+                ' ' * (counter + 4) +
+                '<error message="${testError.name}">';
+            testError.stack.split('\n').forEach((line) {
+              individualTest += '\n' + ' ' * (counter + 6) + line;
+            });
+            individualTest += '\n' + ' ' * (counter + 4) + '</error>';
+          });
+        }
+        individualTest += '\n' + ' ' * (counter + 2) + '</testcase>';
+      }
+      if (test.skipReason != null) {
+        individualTest += '\n' +
+            ' ' * (counter + 4) +
+            '<skipped message="${test.skipReason}"/>';
+      }
+      testResults += '\n' + ' ' * (counter + 2) + individualTest;
+    });
+    return testResults;
+  }
+
+  /// Calculate the test totals for a suite and its children
+  XunitTestSuite _suiteResults(Map group, XunitTestSuite suite) {
+    Map groupStructure = group;
+    XunitTestSuite currentSuite = suite;
+
+    groupStructure.forEach((key, value) {
+      if (key == 'testResults') {
+        value.testResults.forEach((element) {
+          if (element is XunitTestResult) {
+            currentSuite.tests++;
+            if (element.skipped) {
+              currentSuite.skipped++;
+            }
+            if (element.error.isNotEmpty) {
+              if (element.error['failure'] == true) {
+                currentSuite.failed++;
+              } else {
+                currentSuite.errored++;
+              }
+            }
+          }
+        });
+      } else {
+        _suiteResults(groupStructure[key], currentSuite);
+      }
+    });
+
+    return suite;
+  }
+
+  /// Format testsuite headings
+  String _formatTestSuiteHeading(Map group) {
+    XunitTestSuite suite = _suiteResults(group, group['testResults']);
+
+    String suiteHeading = '';
+    if (suite.tests > 0) {
+      suiteHeading += 'tests="${suite.tests}" ';
+    }
+    if (suite.failed > 0) {
+      suiteHeading += 'failures="${suite.failed}" ';
+    }
+    if (suite.errored > 0) {
+      suiteHeading += 'errors="${suite.errored}" ';
+    }
+    if (suite.skipped > 0) {
+      suiteHeading += 'skipped="${suite.skipped}" ';
+    }
+    return suiteHeading.trimRight();
+  }
+
+  /// A method used to create a nested xml hierarchy
+  String _formatXmlHierarchy(Map xmlMap, {int counter: 0}) {
+    counter++;
+    String result = "";
+    xmlMap.keys.forEach((elem) {
+      if (xmlMap[elem] is Map) {
+        var space = ' ';
+        result += space * counter +
+            '<testsuite name="${elem}" ${_formatTestSuiteHeading(xmlMap[elem])}>${_formatTestResults(xmlMap[elem]["testResults"].testResults, counter)}\n';
```

Comment:
you don't need the `{}` for items that are already strings or are using their `toString` method implicitly, so `${elem}` can just be `$elem`.  It's only when you start adding in operators that you need the `{}`